### DISCUSSION TITLE
Platformio upload file

### DISCUSSION THREAD

**Discussion by GurraE** [earlephilhower/arduino-pico#1636] [General]
After a fast reply to my issue regarding "too long filenames" during the installation process, the download now completes fine, and I am able to compile & run the "blink-builtinLED" sketch, by
uploading the file to the PICO via drag & drop of the uf2 file to the pico usb-drive.

After the drag & drop  operation, the pico presents itself as (in this case) COM20.
When trying to upload the file again by using the "upload"-arrow in platformio, the upload starts, then the com-port disappears and the transfer stops with error 1.
 
Is this the correct way of uploading files, (after the first  drag&drop operation), and if so, are there any flags/program lines that has to be appended/edited in platformio.ini to upload files the "normal" way ??

**Comment by LinusHeu** [comment]
check out #1591 

Please tell if you have that same problem. I think there might be more going on than my workaround. I suspect that the version of "picotool" on platformio is too old to just work with the drivers that get auto-installed by windows.

**Reply by GurraE** [reply]
It seems to be the same problem, as Zadig made the magic.
In Zadig my current driver was "none",  so the only option in my case was "Install" instead of "replace".
But it did the trick, upload via platformio/upload arrow now works just fine.
So the next question  is : 
During the upload the comport disappears and after the upload it returns. 
I use the platformio command "Upload & Monitor" and it seems as platformio after the upload is too fast in trying to reestablish the serial port monitor, as I get the message "Could not open port COM20".
After this error message (and when windows pop-up window informs that the serial port is back), I click the  Serial Monitor Icon  and the monitor starts just fine.

So are there any tweaks to get a workaround this problem ??

The commands in my platformio.ini are : 
`[env]
platform = https://github.com/maxgerhardt/platform-raspberrypi.git
framework = arduino
board_build.core = earlephilhower
board_build.filesystem_size = 0.5m

[env:rpipicow]
board = rpipicow
upload_port = COM20
monitor_port = COM20
monitor_speed = 115200
`

**Reply by LinusHeu** [reply]
>  During the upload the comport disappears and after the upload it returns.

That's the expected behaviour. 
It resets the RP2040 to its hardcoded uf2 bootloader, uploads everything, RP2040 reboots and your code starts again.

> after the upload is too fast in trying to reestablish the serial port monitor

I can't reproduce that right now.
A common issue (that afaik can't be fixed by any software on the RP2040) is this:
```ino
void setup(){
Serial.begin();
while(!Serial){}
Serial.println("this might not reliably display because your PC software needs some more time to set the serial monitor up.
In that case, add a delay");
}
```

**Reply by GurraE** [reply]
Yes, I have used this in some projects (not only in Arduino-PICO projects).
When using this, program execution halts until the Serial port is connected,  so its a workaround until I figure out how to instruct platformio to make a short delay after upload, before it connects to the com-port (to the serial monitor)......

Many thanks to Earle Philhower and the rest of you guys for developing and maintaining this project, and for fast replies to all issues.......
